drivers/src/sha256.rs

```diff
@@ -33,6 +35,8 @@ pub trait Sha256Alg {
 
     fn digest_init(&mut self) -> CaliptraResult<Self::DigestOp<'_>>;
     fn digest(&mut self, buf: &[u8]) -> CaliptraResult<Array4x8>;
+    #[cfg(not(feature = "no-cfi"))]
+    fn __cfi_digest(&mut self, buf: &[u8]) -> CaliptraResult<Array4x8>;
```

Comment:
Hmm, this is pretty messy. I wonder if we could use nested functions instead.

Reply:
As is, we need to update other users of this trait (in the fuzz tests)

Reply:
Here's a fairly nice solution:

https://github.com/chipsalliance/caliptra-sw/pull/1060

Probably the sweet spot of complexity vs code size is the second commit in that chain. The third makes it 3x more complex just to save 36 bytes (making the entire PR increase the size by just 4 bytes).

Reply:
Discussed offline; we'll decorate the Crypto::sha256_digest function with CFI instead of the Sha256::digest function.